packages/ra-core/src/i18n/translate.js

```diff
@@ -21,6 +22,15 @@ import wrapDisplayName from 'recompose/wrapDisplayName';
  * @param {*} BaseComponent The component to decorate
  */
 const translate = BaseComponent => {
+    warning(
+        typeof BaseComponent !== 'string',
```

Comment:
how about function components?

Reply:
they are not string right ?

Reply:
warning displays its warning if the condition is **not** met. Here, if `BaseComponent` **is** a string, then we'll have a warning